### DISCUSSION TITLE
MACD instead of EMA crossover strategy

### DISCUSSION THREAD

**Discussion by IHateNameStealers** [TeamWertarbyte/crypto-trading-bot#34] [Ideas]
Is it possible to use MACD instead of the EMA crossover strategy?

I see a MACD indicator in  /react-stockcharts/lib/indicator

Will changing "import { **ema** } from 'react-stockcharts/lib/indicator';" to "import { **macd** } from 'react-stockcharts/lib/indicator';" be enough to make this work?

**Comment by saschb2b** [comment]
Thanks for the suggestion. Tbh I didn't test another indicator yet. 
I assume there has to be change quite a bunch.
 The `CandleStockCharts` type needs to extended to hold the MACD value. Which could be calculated in the `calculateEMA ` method. (Better rename to reflect MACD)
After that you would need to adjust the evaluation happening with the coin here https://github.com/TeamWertarbyte/crypto-trading-bot/blob/master/src/modules/bot/index.ts#L300
There were I check for positive and negative check you would have to inject your logic.

For the open source project it would cool to have the option to use one or another.
This would mean to have a separat function that takes an array of candles and returns a boolean as a buying decision.
PR welcome!

**Comment by saschb2b** [comment]
@IHateNameStealers  Assuming we would have more strategies to choose from than just EMA.
What configurations do you need for MACD?

**Comment by IHateNameStealers** [comment]
I have great success with MACD and FastLength: 12, SlowLength: 26, SignalLength: 9 on HOUR_1
However I think it would be even better on a 15 minute time line!

**Comment by saschb2b** [comment]
Ok interesting. And on what do we make buying/selling decisions? Also crossovers?
Feel free to post any good resources that can help me :)
As I've isolated EMA into a separate class now https://github.com/TeamWertarbyte/crypto-trading-bot/blob/master/src/modules/evaluation/EMAEvaluation.ts it would be now easier to add additional ones

**Comment by IHateNameStealers** [comment]
I buy at a positive MACD crossover, and sell at +3% or at a negative crossover (whatever comes 1st)
My resource is a friend who has been doing this for some time now.
According to him is the "Fibonacci bollinger" strategie also good, however I have no experience with this one.
